libminifi/test/unit/ExpectedTest.cpp

```diff
@@ -484,3 +484,63 @@ TEST_CASE("expected valueOrElse", "[expected][valueOrElse]") {
   REQUIRE_THROWS_AS(ex | utils::valueOrElse([](const std::string&) -> int { throw std::exception(); }), std::exception);
   REQUIRE_THROWS_AS(std::move(ex) | utils::valueOrElse([](std::string&&) -> int { throw std::exception(); }), std::exception);
 }
+
+TEST_CASE("expected transformError", "[expected][transformError]") {
+  auto mul2 = [](int a) { return a * 2; };
+
+  {
+    nonstd::expected<int, int> e = nonstd::make_unexpected(21);
+    auto ret = e | utils::transformError(mul2);
+    REQUIRE(!ret);
+    REQUIRE(ret.error() == 42);
+  }
+
+  {
+    const nonstd::expected<int, int> e = nonstd::make_unexpected(21);
+    auto ret = e | utils::transformError(mul2);
+    REQUIRE(!ret);
+    REQUIRE(ret.error() == 42);
+  }
+
+  {
+    nonstd::expected<int, int> e = nonstd::make_unexpected(21);
+    auto ret = std::move(e) | utils::transformError(mul2);
+    REQUIRE(!ret);
+    REQUIRE(ret.error() == 42);
+  }
+
+  {
+    const nonstd::expected<int, int> e = nonstd::make_unexpected(21);
+    auto ret = std::move(e) | utils::transformError(mul2);  // NOLINT(performance-move-const-arg)
+    REQUIRE(!ret);
+    REQUIRE(ret.error() == 42);
+  }
```

Comment:
I think `clang-tidy` has a point here: this test doesn't make much sense, since we can't move from a `const` variable.  I would remove this test case, and the others marked with `NOLINT(performance-move-const-arg)`, too.  Maybe they could be replaced with more realistic tests where the `expected` comes from the return value of a function.

Reply:
It doesn't make much sense to use const rvalue refs, but this still checks that the implementation matches the expectations in this unlikely case. I just copy pasted the transform tests from above, which were copy pasted from TartanLlama's test suite.

How would you structure the suggested new test case, that ends up testing the expected implementation, and not the C++ language itself?

Reply:
As it is, this is the same test as the one at lines 499-502, just with a confusing `move` added which doesn't do anything.

I think something like this would be slightly different from lines 506-509:
```c++
  {
    const auto expected_provider = [] { return = nonstd::expected<int, int>(nonstd::unexpected, 21); };
    auto ret = expected_provider() | utils::transformError(mul2);
    REQUIRE(!ret);
    REQUIRE(ret.error() == 42);
  }
```
but if you don't think it's different enough, I'm OK with removing this test case.

Reply:
I don't mind replacing the mutable rvalue case with this, since it's what is being tested in your example, but I'd still keep the const rvalue case, with the NOLINT exclusion, to keep a full coverage of possible input reference types.